polaris/polaris/utils.py

```diff
@@ -135,24 +135,23 @@ def create_stellar_deposit(
         transaction.save()
         raise ValueError(transaction.status_message)
 
-    # if we don't know if the destination account exists
-    if not destination_exists:
-        try:
-            _, created, pending_trust = get_or_create_transaction_destination_account(
-                transaction
-            )
-        except RuntimeError as e:
-            transaction.status = Transaction.STATUS.error
-            transaction.status_message = str(e)
+    # check if the account is doesn't trust  the asset to be received
+    try:
+        _, json_resp = get_account_obj(
+            Keypair.from_public_key(transaction.stellar_account)
+        )
+    except RuntimeError as e:
+        transaction.status = Transaction.STATUS.error
+        transaction.status_message = str(e)
+        transaction.save()
+        logger.error(transaction.status_message)
+        return False
+    if has_trustline(transaction, json_resp):
+        # the account is doesn't trust the asset to be received
+        if transaction.status != Transaction.STATUS.pending_trust:
+            transaction.status = Transaction.STATUS.pending_trust
```

Comment:
Now that I'm seeing the code change, I'm realizing it may be unnecessary to split up the removal of the `pending_trust` logic. Because if we remove the `get_or_create_transaction_destination_account()` invocation AND the `pending_trust` logic in one change, we can just remove this entire section right?

Reply:
And if we're removing the `pending_trust` recovery functionality, we also do it in `submit_stellar_deposit()`, so we'll need to remove it there too.

Reply:
I don't think I understand, remove the entire section try block or     
```
if has_trustline(transaction, json_resp):
        # the account is doesn't trust the asset to be received
        if transaction.status != Transaction.STATUS.pending_trust:
            transaction.status = Transaction.STATUS.pending_trust
```

Reply:
If we remove that section you just referenced, do we need the try block?

Reply:
So in that case we should remove the test `def test_deposit_stellar_no_account(acc1_usd_deposit_transaction_factory):` since it serves no plausible scenario where the account doesnt exist and must be created.

Reply:
I will have to put some of this logic back in if I want to tell if the given account has a `trustline` and is `claimable_balance_supported`.